Let know to lazy relation wrapper that entity was loaded using query builder

serega3000 opened the pull request:
Maked for lazy relations related entity to be not loaded again after selecting with join and fetching it from current entity. Added `__has__*=true` to RawSqlResultsToEntityTransformer.
But then some tests make failed, and i founded that if we have inverse side relation, ex. category.posts <-> post.category, when we fetching post from category (category.posts), categories (post.categories) will be also added in result set with empty array, because it persist in join method in LazyRelationWrapper, and it will be added as alias to query builder and then to RawSqlResultsToEntityTransformer. The only solution i found is to remove `__has__*` value from result object in LazyRelationWrapper.

pleerock commented:
Hi and thank you for your PR!

Please make sure to add tests for the functionality you add.

Sorry but I didn't understood the reason why you added `remove __has__*` code into `LazyRelationWrapper`. Maybe you can comment that code and add some tests which shows the problem?

serega3000 commented:
Ok, i will add tests and comments.
I try to explain why i added `remove __has__*` into `LazyRelationWrapper`, may be exists better solution, but i did not founded it.

When i just added `entity["__has__" + propertyName + "__"] = true;` tests failed on `github issues > #234 and #223 lazy loading does not work correctly from one-to-many and many-to-many sides`

I made some debugging and understand problem:
For example, we have lazy relations and inverse side relation, `category.posts` <-> `post.category`.
We have variable `post`
we try to get category from post `let category = await post.category`
and then we want to get all posts in this category `let posts = await  category.posts`.
We always will get empty array.

It is because when `LazyRelationWrapper` creating query for finding category, he also joining post. https://github.com/typeorm/typeorm/blob/master/src/lazy-loading/LazyRelationsWrapper.ts#L91
When post table joined, it added to `AliasMap` in `QueryBuilder`
Then `QueryBuilder` send `AliasMap` to `RawSqlResultsToEntityTransformer`
Next, when `RawSqlResultsToEntityTransformer` build result, line 163:
https://github.com/typeorm/typeorm/blob/master/src/query-builder/transformer/RawSqlResultsToEntityTransformer.ts#L163

` const relationAlias = this.aliasMap.findAliasByParent(alias.name, relation.propertyName);`
relation alias of 'post' founded.


`const relatedEntities = this.groupAndTransform(rawSqlResults, relationAlias);`
returns empty array, although no one column of post selected in query


```
const isResultArray = relation.isManyToMany || relation.isOneToMany;
const result = !isResultArray ? relatedEntities[0] : relatedEntities;
```
result still empty array


```
                if (result) {
                    ....
                    if (relation.isLazy) {
                        entity["__" + propertyName + "__"] = result;
                        entity["__has__" + propertyName + "__"] = true;
                    } else {
                   .....
```
Result of empty posts will be added to category, and when we try to get posts from this category, we always get empty array.

pleerock commented:
@serega3000 Im sorry for late response here, but are you still interested in this fix?

pleerock commented:
If you are please check latest version if your issue is already fixed. If its not then please create a test that reproduce your issue and I'll make a fix by myself.
